### PR TITLE
Add pylint-pytest plugin

### PR DESCRIPTION
**Pull request by drew2a**
This PR adds the `pylint-pytest` plugin to the `.pylintrc` file, enabling additional checkers for pytest. It also includes the `pylint-pytest` package in the `requirements-test.txt` file, ensuring it is installed when running tests.

The necessity of adding this plugin emerged from the notable discussion in #7764, where @drew2a proposed the use of `pylint-pytest` and @qstokkink supported the idea:

>Just to repeat - myself and the docs - adding a "name" is not the only way to resolve this. Secondarily, this has nothing to do with Pylint, other than that it will also tell you the same thing. I like the idea of adding pylint-pytest though.

Ref:
 - https://github.com/Tribler/tribler/pull/7764#pullrequestreview-1774459402
 - https://github.com/pylint-dev/pylint-pytest

### CODE REVIEW

**Comment by drew2a** [comment]
Unfortunately, it seems not possible to use this plugin with Codacy, as they use preinstalled plugins:
 - https://docs.codacy.com/repositories-configure/configuring-code-patterns/#import-patterns
 - https://github.com/codacy/codacy-pylint-python3/blob/master/requirements.txt

**Comment by drew2a** [comment]
I've created a PR to `codacy-pylint-python3` to extend the plugin list:
- https://github.com/codacy/codacy-pylint-python3/pull/196

Let's wait for the response from the repository maintainers.

**Comment by drew2a** [comment]
The repository maintainers responded that the requested change was added:
- https://github.com/codacy/codacy-pylint-python3/pull/197

**Review by egbertbouman — APPROVED** [review]

**Review by xoriole — APPROVED** [review]

**Review by kozlovsky — APPROVED** [review]

**Pull request merged**
